src/part_file.cpp

```diff
@@ -314,9 +314,6 @@ namespace libtorrent
 
 				if (!buf) buf.reset(new char[m_piece_size]);
 
-				// don't hold the lock during disk I/O
-				l.unlock();
```

Comment:
For me it's more like blocking i/o could be neglectable here, because the only case for this is changing file priorities.
No other operations would suffer from this change.

Reply:
in principle, the `readv()` function could break for the same reason. Imagine you have one thread in a `readv()` call, where the partfile was opened in read-only. Another thread comes around and wants to *write* to the partfile. It will then re-open the file in read-write mode. Possibly breaking the outstanding read operation in the first thread.

Reply:
this is a separate issue though, and should be fixed in a separate PR I think